examples/map.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey"
+)
+
+// the questions to ask
+var simpleQs = []*survey.Question{
+	{
+		Name: "name",
+		Prompt: &survey.Input{
+			Message: "What is your name?",
+		},
+		Validate: survey.Required,
+	},
+	{
+		Name: "color",
+		Prompt: &survey.Select{
+			Message: "Choose a color:",
+			Options: []string{"red", "blue", "green"},
+		},
+		Validate: survey.Required,
+	},
+}
+
+func main() {
```

Comment:
I'd also like to avoid the additional argument to `AskOne` as that would introduce an API breaking change. I really appreciate all the effort and hard work you've put into these changes, but I dont think I want to incorporate them just yet. I'll need to do a lot of thinking about the "role" of survey and what sorts of things to push as use-case specific layers on top.